src/jquery.contextMenu.js

```diff
@@ -1411,6 +1429,35 @@
                 }
 
                 return $layer;
+            },
+            processPromises: function (opt, root, promise) {
+                function completedPromise(opt,root,items) {
+                    //completed promise (dev called promise.resolve)
+                    //we now have a list of items which can be used to create the rest of the context menu.
+                    if (items === undefined) {
+                        //meh, null result, dev should have checked
+                        errorPromise(undefined);//own error object
+                    }
+                    finishPromiseProcess(opt,root, items);
+                };
+                function errorPromise(opt,root,errorItem) {
+                    //user called promise.reject() with an error item, if not, provide own error item.
+                    if ( typeof errorItem === 'string'  || errorItem === undefined) {
+                        errorItem = { "error": { name: "No items and no error item", icon: "context-menu-icon context-menu-icon-quit" } };
```

Comment:
Hmm. I might had a reason for that but not sure why anymore, maybe to force users to build nice error items or something. I'll update it, one second.